boa_engine/src/context/mod.rs

```diff
@@ -709,10 +714,21 @@ impl Context {
         self.realm.set_global_binding_number();
         let result = self.run();
         self.vm.pop_frame();
+        self.run_queued_jobs();
```

Comment:
one thing that confused me, is in your example i would have expected `count` to be three. Because execute waits here for the promise job to finish before finishing. But it seems that's not the case.  It looks like it returns 2 and you need to call it again (looking at your test).

Can someone explain that?

Reply:
The `Promise` constructor executes its code on creation, and `then` is queued on the job queue. After executing all the statements of the script, `count === 2` and `then` is executed, making `count === 3`. However, the script already returned `2` as the evaluation result, since `let result = self.run()` executes before `self.run_queued_jobs()`. When the next execution occurs, `count` is already 3, so the evaluation returns 3.